### PR TITLE
fix Average metric

### PR DESCRIPTION
**Pull request by epwalsh**
Closes https://github.com/allenai/allennlp/issues/4623.

There was a bug with the `average` and `evalb_bracketing_scorer` metrics during distributed training on GPU.

### CODE REVIEW

File: allennlp/common/testing/distributed_test.py [hunk 1]
```diff
@@ -69,5 +69,5 @@ def run_distributed_test(
         init_process,
         args=(device_ids, world_size, func, args, kwargs),
         nprocs=nprocs,
-        start_method="fork",
+        start_method="spawn",
```
Comment (epwalsh) [inline allennlp/common/testing/distributed_test.py:72]: Need to use spawn as the start method in order to run these tests on GPU for some reason.

Comment (AkshitaB) [reply allennlp/common/testing/distributed_test.py:72]: Fwiw, using the `spawn` method was causing some pickle-related error, when I was working on this. https://github.com/pytorch/pytorch/issues/22585

Comment (epwalsh) [reply allennlp/common/testing/distributed_test.py:72]: Hmm darn! Ok, I'll take another look

Comment (epwalsh) [reply allennlp/common/testing/distributed_test.py]: Okay this is ready for another look @AkshitaB

**Review by AkshitaB — APPROVED** [review]
LGTM!

**Pull request merged**